cpp/README.md

```diff
@@ -0,0 +1,62 @@
+# Lance: A Columnar File Format
```

Comment:
Top level says data format and this says file format

Reply:
Fixed. To be all data format.